### PR TITLE
[HIVEMALL-124][BUGFIX] Fixed bugs in BinaryResponseMeasure (nDCG, MRR, AP)

### PR DESCRIPTION
**Pull request by myui**
## What changes were proposed in this pull request?

Fixed bug in the BinaryResponse measure such as nDCG, MRR, (M)AP.

## What type of PR is it?

Bug Fix

## What is the Jira issue?

https://issues.apache.org/jira/browse/HIVEMALL-124

## How was this patch tested?

unit tests, manual tests

## Checklist

- [x] Did you apply source code formatter, i.e., `mvn formatter:format`, for your commit?
- [x] Did you run manual test on Hive/Spark CLI on the modified functions?

### CODE REVIEW

**Comment by myui** [comment]
@takuti could you review this PR?

File: core/src/main/java/hivemall/evaluation/BinaryResponsesMeasures.java [hunk 1]
```diff
@@ -120,48 +148,65 @@ public static int countTruePositive(final List<?> rankedList, final List<?> grou
...
-                sumPrecision += nTruePositive / (i + 1.0);
+                sumPrecision += nTruePositive / (i + 1.d);
             }
         }
 
-        return sumPrecision / groundTruth.size();
+        if (nTruePositive == 0) {
+            return 0.d;
+        }
+        return sumPrecision / nTruePositive;
```
Comment (takuti) [inline core/src/main/java/hivemall/evaluation/BinaryResponsesMeasures.java]: 👍

**Comment by myui** [comment]
Ideally, it should be `ndcg(array(1,3), array(1)) < ndcg(array(1), array(1))` but correct code return 1.0 for both cases.

File: core/src/main/java/hivemall/evaluation/BinaryResponsesMeasures.java [hunk 2]
```diff
@@ -79,8 +91,15 @@ public static double IDCG(final int n) {
...
-            @Nonnull final List<?> groundTruth, @Nonnull final int recommendSize) {
-        return (double) countTruePositive(rankedList, groundTruth, recommendSize) / recommendSize;
+            @Nonnull final List<?> groundTruth, @Nonnegative final int recommendSize) {
+        if (rankedList.isEmpty()) {
+            if (groundTruth.isEmpty()) {
+                return 1.d;
+            }
+            return 0.d;
+        }
+        final int k = Math.min(rankedList.size(), recommendSize);
```
[the remark this answers is not in the document]
Comment (myui) [reply core/src/main/java/hivemall/evaluation/BinaryResponsesMeasures.java]: Fixed in https://github.com/apache/incubator-hivemall/pull/115/commits/759b088370fb9e6b8d17c46dd9d71f7c1980a78a

Recall is as it is.

**Pull request closed without merging**